addons/website_mail/static/src/js/website_mail.js

```diff
@@ -0,0 +1,193 @@
+                // bind events
+                self.on("change:messages", self, self.render_messages);
+                self.on("change:message_count", self, function(){
+                    self.render_message_count();
+                    self.set('pager', self._pager(self.get('current_page')));
+                });
+                self.on("change:pager", self, self.render_pager);
+                self.on("change:current_page", self, self.on_change_current_page);
+                self.on("change:domain", self, self.on_change_domain);
+                // set options and parameters
+                self.options = _.extend(self.options, result['options'] || {});
+                self.set('message_count', self.options['message_count']);
+                self.set('messages', self.preprocess_messages(result['messages']));
+            });
+        },
+        _load_templates: function(){
+            return ajax.loadXML('/website_mail/static/src/xml/website_mail.xml', qweb);
+        },
+        fetch_message: function(domain){
+            var self = this;
+            var data = this._prepare_fetch_message_params();
+            return ajax.jsonRpc('/website_mail/fetch', 'call', data).then(function(result){
+                self.set('messages', self.preprocess_messages(result['messages']));
+                self.set('message_count', result['message_count']);
+            });
+        },
+        _prepare_fetch_message_params: function(){
+            var self = this;
+            var data = {
+                'res_model': this.options['res_model'],
+                'res_id': this.options['res_id'],
+                'limit': this.options['pager_step'],
+                'offset': (this.get('current_page')-1) * this.options['pager_step'],
+                'allow_composer': this.options['allow_composer'],
+            }
+            // add fields to allow to post comment without being logged
+            _.each(['token', 'token_field'], function(field){
+                if(self.options[field]){
+                    data[field] = self.options[field];
+                }
+            });
+            // add domain
+            if(this.get('domain')){
+                data['domain'] = this.get('domain');
+            }
+            return ajax.jsonRpc('/website_mail/fetch', 'call', data).then(function(result){
+                self.set('messages', self.preprocess_messages(result['messages']));
+                self.set('message_count', result['message_count']);
+            });
+        },
+        preprocess_messages: function(messages){
+            _.each(messages, function(m){
+                m['author_avatar_url'] = _.str.sprintf('/web/image/%s/%s/author_avatar/50x50', 'mail.message', m.id);
+                m['published_date_str'] = _.str.sprintf(_t('Published on %s'), moment(m.date).format('MMMM Do YYYY, h:mm:ss a'));
+            });
+            return messages;
+        },
+        _pager: function(page, total, step, scope){
```

Comment:
I would either remove parameters or explicitely set them to this.get(parameter) when calling the function, to lessen the number of variable processing.